RoaringBitmap/src/main/java/org/roaringbitmap/longlong/Roaring64NavigableMap.java

```diff
@@ -692,6 +692,51 @@ private int highestHigh() {
     return RoaringIntPacking.highestHigh(signedLongs);
   }
 
+  // to be used with lazyor
+  public void repairAfterLazy() {
+    for (Entry<Integer, BitmapDataProvider> e2 : highToBitmap.entrySet()) {
+      BitmapDataProvider lowBitmap2 = e2.getValue();
+      if (lowBitmap2 instanceof RoaringBitmap) {
+        ((RoaringBitmap) lowBitmap2).repairAfterLazy();
+      } else {
+        throw new UnsupportedOperationException(
```

Comment:
It is likely unnecessary to throw an exception here since naivelazyor also only operate on RoaringBitmap instances. So I think you can slightly simplify the code.

Reply:
But the rest of the class is written with these exceptions so you were probably right in having them.